src/utils/offlineRequestHandler.ts

```diff
@@ -156,12 +159,15 @@ export class OfflineRequestHandler {
         request.onerror = (event) => {
           const error = (event.target as IDBRequest<IDBDatabase>).error;
           if (
-            error?.name === 'SecurityError' ||
-            error?.message?.includes('denied permission')
+            INDEXEDDB_ACCESS_DENIED_ERRORS.some((msg) =>
+              error?.message?.includes(msg)
+            )
           ) {
-            reject(new IndexedDBError('IndexedDB access denied'));
+            console.warn('IndexedDB access denied:', error);
+            resolve(undefined);
```

Comment:
the resolve(undefined) can be pulled out of the if-else statement and called at the very end